app/frontend/src/addVacancyStateToDataLayer.js

```diff
@@ -0,0 +1,15 @@
+export const addVacancyStateToDataLayer = {
+  getVacancyDataSet() {
+    const element = document.querySelector('.new_copy_vacancy_form') || document.body
+    return element.dataset;
+  },
+
+  getObjectForDataLayer() {
+    const vacancyDataset = this.getVacancyDataSet();
+    return { vacancy_state: vacancyDataset.vacancyState };
+  }
+}
+
```

Comment:
This is a very ruby way to do it. A more JS way would be:

```
export const default = document.addEventListener("DOMContentLoaded", function(e) {
  const element = document.querySelector('.new_copy_vacancy_form') || {};
  const dataset = element.dataset || {};
  const vacancy_state = dataset.vacancyState;
  dataLayer.push({ vacancy_state });
});
```
I'm pretty sure you don't need the `export const default = ` either, but I'm showing it in case you get console errors.

Reply:
I'm OK with it being rubish, but the extra function calls add unnecessary overhead and you don't need the `export` as all that does is expose the exported function(s) to the importing lib.  All the work gets done the moment the script gets hit.  I'll leave it up to you if you want to adapt this.  If you stay with this way, let me know and I'll put a snagging ticket in for later.